src/modules/todo/components/todo/Todo.tsx

```diff
@@ -1,34 +1,33 @@
-import type { TodoItem } from './Todo.interface'
 import { useForm } from 'react-hook-form'
-import { useFetchTodos } from '../../../../hooks/useFetchTodos'
 import * as classes from './Todo.module.scss'
+import { useMutationToPostTodo } from '../../hooks/useMutationToPostTodo'
+import { useFetchTodos } from '../../hooks/useFetchTodos'
 
 type AddTodoFormValues = {
   text: string
 }
 
 function Todo() {
-  const [todos, setTodos] = useFetchTodos()
-
   const { register, handleSubmit, reset } = useForm<AddTodoFormValues>()
 
-  const onSubmit = handleSubmit((data) => {
-    fetch('/api/todos', { method: 'POST', body: JSON.stringify(data) })
-      .then(async (res) => {
-        const { todo } = (await res.json()) as { todo: TodoItem }
-        setTodos((prev) => [...prev, todo])
-        reset()
-      })
-      .catch((err) => {
-        console.error(err)
-      })
+  const todoList = useFetchTodos()
+
+  const mutation = useMutationToPostTodo()
+
+  const onSubmit = handleSubmit(({ text }) => {
+    mutation.mutate({
+      text,
+    })
+
+    reset()
   })
 
   return (
     <div>
       <h1 className={classes.root}>Todo list</h1>
       <ul>
-        {todos.map((todo) => {
+        {todoList.isLoading && 'loading...'}
```

Comment:
Instead of just showing 'loading' let's make a wrapper that will accept loading param and children and display loading on top of children if we have some, later we can improve it.